README.md

```diff
@@ -172,9 +174,53 @@ Can you fetch recent events from our "production-cluster"
 to help identify what might be causing these deployment issues?
 ```
 
+## Troubleshooting
+
+### Authentication Issues
+
+In most cases, something is wrong with the OAuth Client. Please ensure that you have added all scopes as requested above.
+In addition, please ensure that your user also has all necessary permissions on your Dynatrace Environment.
+
+In case of any problems, you can troubleshoot SSO/OAuth issues based on our [Dynatrace Developer Documentation](https://developer.dynatrace.com/develop/access-platform-apis-from-outside/#get-bearer-token-and-call-app-function) and providing the list of scopes.
+
+It is recommended to try access the following API (which requires minimal scopes `app-engine:apps:run` and `app-engine:functions:run`):
+
+1. Use OAuth Client ID and Secret to retrieve a Bearer Token (only valid for a couple of minutes):
+```bash
+curl --request POST 'https://sso.dynatrace.com/sso/oauth2/token' \
+  --header 'Content-Type: application/x-www-form-urlencoded' \
+  --data-urlencode 'grant_type=client_credentials' \
+  --data-urlencode 'client_id={your-client-id}' \
+  --data-urlencode 'client_secret={your-client-secret}' \
+  --data-urlencode 'scope=app-engine:apps:run app-engine:functions:run'
+```
+
+2. Use `access_token` from the response of the above call as the bearer-token in the next call:
+```bash
+curl -X GET https://abc12345.apps.dynatrace.com/platform/management/v1/environment \
+  -H 'accept: application/json' \
+  -H 'Authorization: Bearer {your-bearer-token}'
+```
+
+3. You should retrieve a result like this:
+```json
+{
+  "environmentId": "abc12345",
+  "createTime": "2023-01-01T00:10:57.123Z",
+  "blockTime": "2025-12-07T00:00:00Z",
+  "state": "ACTIVE"
+}
+```
+
+
+### Problem accessing data on Grail
+
+Grail has a dedicated section about permissions in the Dynatrace Docs. Please refer to https://docs.dynatrace.com/docs/discover-dynatrace/platform/grail/data-model/assign-permissions-in-grail for more details.
```

Comment:
Minor suggestion (it's not going to make or break it): SeaOtter link for documentation might be interesting so these URLs can be amended without additional PRs, and are still valid in older commits if somebody looks at one for whichever reason.

Reply:
As this is an open-source repo where everyone (not just Dynatracers) can contribute, but only a handful people can create SeaOtter (aka dt-url.net) shortlinks, I opted against that.